packages/functions/src/coordinator/GetCoordinatorAppointmentsHandler.ts

```diff
@@ -28,11 +31,69 @@ export default async function (
     }
   });
 
-  const donorsInAppointments = await getDonors(donorIds);
+  let donorsInAppointments = await getDonors(donorIds);
+
+  if (coordinator.role === CoordinatorRole.GROUP_COORDINATOR) {
+    donorsInAppointments = await filterDonorsInGroup(
+      coordinator,
+      donorsInAppointments
+    );
+    appointments = filterAppointmentsForDonors(
+      appointments,
+      donorsInAppointments
+    );
+  }
 
   const res: FunctionsApi.GetCoordinatorAppointmentsResponse = {
     appointments,
     donorsInAppointments,
   };
   return res;
 }
+
+async function fetchCoordinator(callerId: string) {
+  const coordinator = await getCoordinator(callerId);
+  if (!coordinator) {
+    console.error("Could not find calling user", callerId);
+    throw Error("User is not an coordinator and can't edit appointments");
+  }
+  return coordinator;
+}
+
+async function validate(coordinator: DbCoordinator, hospital: Hospital) {
+  switch (coordinator.role) {
+    case CoordinatorRole.SYSTEM_USER:
+      return;
+    case CoordinatorRole.ZM_COORDINATOR:
+    case CoordinatorRole.HOSPITAL_COORDINATOR:
+      if (!coordinator.hospitals?.includes(hospital)) {
+        console.error(
+          `Coordinator ${coordinator.id} ${coordinator.role} ${coordinator.hospitals} missing permissions for ${hospital}`
+        );
+        throw Error("Coordinator has no permissions for hospital");
+      }
+      break;
+    case CoordinatorRole.GROUP_COORDINATOR:
+      break;
+  }
+}
+
+async function filterDonorsInGroup(
+  coordinator: DbCoordinator,
+  donorsInAppointments: DbDonor[]
+) {
+  const groupIds = await GroupDAL.getGroupIdsOfCoordinatorId(coordinator.id);
```

Comment:
Should this be Set?

Reply:
Done
There are many places where we should use sets but use arrays.